contracts/SpokePool.sol

```diff
@@ -147,19 +182,73 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
     }
 
     function initiateRelay(
-        uint256 originChain,
-        address sender,
+        uint256 originChainId,
         uint256 amount,
-        address recipient,
-        uint256 relayerFee,
-        uint256 realizedLpFee
-    ) public {}
+        uint64 depositId,
+        uint64 relayerFeePct,
+        uint64 realizedLpFeePct,
+        address destinationToken,
+        address sender,
+        address recipient
+    ) public {
+        // We limit the relay fees to prevent the user spending all their funds on fees.
+        require(relayerFeePct <= 0.5e18 && realizedLpFeePct <= 0.5e18, "invalid fees");
+
+        // Use new relay count as unique ID for new relay.
+        relays.push(
+            RelayData(
+                recipient,
+                destinationToken,
+                relayerFeePct,
+                realizedLpFeePct,
+                amount, // total relay amount
+                0 // total amount filled
+            )
+        );
+        uint64 relayId = uint64(relays.length - 1);
```

Comment:
do you think it will use less gas to not define this variable at all and rather just put `uint64(relays.length - 1)` on line 214?

Reply:
Yeah it should ill change